### ISSUE TITLE
two full step re-renders on field change

### ISSUE THREAD

**Issue by Cooryd** [elifesciences/elife-xpub#2088]
**Describe the bug**
When triggering an onChange on a form field (input, checkbox ect.) the whole step re-renders twice. This should be optimised. Check the props being passed down to the Step `formikProps` and only pass those needed.

**Comment by diversemix** [comment]
Part of #2064

**Comment by Cooryd** [comment]
What I've found so far:
https://github.com/jaredpalmer/formik/issues/342#issuecomment-489346081
https://travix.io/using-recompose-hocs-to-get-better-performance-in-forms-using-formik-and-yup-e51024d645ba
https://github.com/armand1m/create-formik-fields-comparator

**Comment by Cooryd** [comment]
So the code snippet for checking if a field should rerender looks like this 

```const createFormikFieldsComparator = (fields: ReadonlyArray<string>) => (
  current: Props,
  next: Props
): boolean => {
  const compare = (type: PossibleFieldTypes) => (field: string) =>
    current[type][field] !== next[type][field];

  const test = (type: PossibleFieldTypes) => fields.some(compare(type));

  return test('values') || test('errors') || test('touched');
};

export default createFormikFieldsComparator;```

Appears to just diff the fields property based on its value in the ‘values’, ‘touched’ and ‘errors’ props. I’ll look at creating a custom hook that implements this

**Comment by Cooryd** [comment]
Looks like `Formik` isn't the perpetrator here. The `withApollo` generated props seem to cause the step components to re-render when ever there is a user interaction ie: focusing on inputs ????

I've diffed the `prevProps ` and `currentProps` and it looks like the following are causing the re-render:

```
data: {
fetchMore: ƒ ()
refetch: ƒ ()
startPolling: ƒ ()
stopPolling: ƒ ()
subscribeToMore: ƒ ()
updateQuery: ƒ ()
}
```

re-focusing on inputs should be handled locally by the input components so I'm unsure why this is causing Apollo based props to change, especially this far up the render tree. Will look into any Apollo issues raised to see if this is a known bug? We will move too Apollo with hooks once it has been released. 

The best solution I can think of is to memoize the data prop comparison for GQL wrapped components to not diff these properties as they are all functions which should remain static.
